python/pfs/drp/stella/fitPfsFluxReference.py

```diff
@@ -0,0 +1,978 @@
+        (variance = (median(flux) / snr)**2).
+        Note that no actual noise will be added to the input flux.
+
+    Returns
+    -------
+    fiberArraySet : `pfs.datamodel.pfsFiberArraySet.PfsFiberArraySet`
+        `PfsFiberArraySet` that contains only the input fiber.
+    """
+    observations = Observations(
+        visit=np.zeros(0, dtype=int),
+        arm=np.zeros(0, dtype="U0"),
+        spectrograph=np.zeros(0, dtype=int),
+        pfsDesignId=np.zeros(0, dtype=int),
+        fiberId=np.zeros(0, dtype=int),
+        pfiNominal=np.zeros(shape=(0, 2), dtype=float),
+        pfiCenter=np.zeros(shape=(0, 2), dtype=float)
+    )
+
+    spectrum = PfsFiberArray(
+        target=spectrum.target,
+        observations=observations,
+        wavelength=spectrum.wavelength,
+        flux=spectrum.flux,
+        mask=spectrum.mask,
+        sky=np.zeros(shape=spectrum.flux.shape, dtype=float),
+        covar=np.zeros(shape=(3,) + spectrum.flux.shape, dtype=float),
+        covar2=np.zeros(shape=(0, 0), dtype=float),
+        flags=spectrum.flags,
+        metadata=spectrum.metadata,
+    )
+
+    noise = np.nanmedian(spectrum.flux) / snr
+    spectrum.covar[0, :] = noise**2
+
+    return spectrum
+
+
+def promoteFiberArrayToFiberArraySet(spectrum, fiberId):
+    """Promote an instance of PfsFiberArray to PfsFiberArraySet.
+
+    Parameters
+    ----------
+    spectrum : `pfs.datamodel.pfsFiberArray.PfsFiberArray`
+        spectrum observed with a fiber.
+    fiberId : `int`
+        ID of the fiber.
+        .
+    Returns
+    -------
+    fiberArraySet : `pfs.datamodel.pfsFiberArraySet.PfsFiberArraySet`
+        `PfsFiberArraySet` that contains only the input fiber.
+    """
+    return PfsFiberArraySet(
+        identity=Identity(visit=0, arm="", spectrograph=1, pfsDesignId=0),
+        fiberId=np.full(shape=(1,), fill_value=fiberId, dtype=int),
+        wavelength=spectrum.wavelength.reshape(1, -1),
+        flux=spectrum.flux.reshape(1, -1),
+        mask=spectrum.mask.reshape(1, -1),
+        sky=spectrum.sky.reshape(1, -1),
+        norm=np.ones(1, dtype=float),
+        covar=spectrum.covar.reshape((1,) + spectrum.covar.shape),
+        flags=spectrum.flags,
+        metadata=spectrum.metadata,
+    )
+
+    return spectrum
+
+
+def fibers(pfsConfig, fiberArraySet):
+    """Iterator that yields each fiber in `PfsFiberArraySet`.
+
+    Parameters
+    ----------
+    pfsConfig : `pfs.datamodel.pfsConfig.PfsConfig`
+        Configuration of the PFS top-end.
+    fiberArraySet : `pfs.datamodel.pfsFiberArraySet.PfsFiberArraySet`
+        Set of spectra observed with a set of fibers
+
+    Yields
+    -------
+    fiberArray : `pfs.datamodel.pfsFiberArray.PfsFiberArray`
+        spectrum observed with a fiber.
+    """
+    for fiberId in pfsConfig.fiberId:
+        yield fiberArraySet.extractFiber(PfsFiberArray, pfsConfig, fiberId)
+
+
+def fiberConfigs(pfsConfig):
+    """Iterator that yields single-fiber PfsConfigs.
```

Comment:
I think there's a note in `PfsConfig` saying that "we're not going to provide an iterator because it would be inefficient". Maybe it's not so bad, and this should live there?

Reply:
I agree that this iteration is inefficient, but its inefficiency is negligible in `FitPfsFluxReferenceTask`. The iterators (`fiberConfigs()` and `fibers()`) are convenient indeed.

Reply:
I don't have a strong opinion whether `PfsConfig` itself should be iterable nonetheless for the inefficiency.